docs: Add "Notifications" page (#69)
phylum-dev/documentation: docs/knowledge_base/notifications.md

## Notification Setup

### Email Notifications

1. Navigate to the 'Notifications' section in your Phylum group or project.
2. Enter the email addresses that you wish to receive notifications.
3. `Confirm` the addresses that will be sent notifications.
4. A `Subscription Confirmation` email will be sent and each user will need to confirm their subscription.

### Webhook Integration

1. Navigate to the 'Notifications' section in your Phylum group or project.
2. Follow the respective setup guide for the currently supported platforms:
   - **Slack**: [Slack Webhook Setup](https://api.slack.com/messaging/webhooks#getting_started)
   - **Microsoft Teams**: [Microsoft Teams Webhook Setup](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook)
   - **Discord**: [Discord Webhook Guide](https://discord.com/developers/docs/resources/webhook#execute-slackcompatible-webhook) (Note: Discord uses a Slack-compatible webhook format)
3. Enter the webhook URL provided by your chosen platform.
4. `Update` your settings to enable Webhook notifications.

## Conclusion

> Setting up notifications for emails or webhooks ensures you stay updated with your Phylum account activities. Follow these steps to receive notifications via your preferred communication method.

Reviewer: Just wondering if we should mention something about the option to pause notifications

Reviewer 2: @ejortega we likely should mention it, mostly just to spell out exactly what will happen. Does it only pause project level settings or also include group notifications, how are notifications resumed, etc.

Reviewer 2: resolving for now, we can add this in later when you are feeling better

---

> title: Setting up Notifications in Phylum

Reviewer 2: ```suggestion
title: Notifications
```

Shortening this title for `readme.com` display purposes

---
title: Notifications
category: 6255e67693d5200013b1fa41
hidden: false
---

# Overview

Phylum supports notification systems using email addresses or webhooks for various platforms including Slack, Microsoft Teams, and Discord. This document guides you through setting up notifications for each of these platforms.

> Notifications are a Phylum Pro feature and can be configured at the group or project level. Currently, the only trigger for notifications is [continuous monitoring](https://docs.phylum.io/docs/continuous_monitoring) events.

Reviewer 3: ```suggestion
Notifications are a Phylum Pro feature and can be configured at the group or project level. Currently, the only trigger for notifications is [Continuous Monitoring](https://docs.phylum.io/docs/continuous_monitoring) events.
```

## Notification Setup

### Email Notifications

1. Navigate to the 'Notifications' section in your Phylum group or project.

> 2. Enter the email address where you wish to receive notifications.

Reviewer: perhaps we could indicate here that a comma-separated list of email addresses is also an option

> 3. Save your settings to enable email notifications.

Reviewer 3: They're not really "saving settings", they are sending invites.

Reviewer 2: @mathew-horner looks updated, is this resolved?

### Slack Webhook Integration

1. Create a Slack webhook following the guide at [Slack Webhook Setup](https://api.slack.com/messaging/webhooks#getting_started).
2. In Phylum, navigate to the 'Notifications' section.
3. Enter your Slack webhook URL.

> 4. Perform a test to ensure the notifications are being sent correctly.

Reviewer 3: They can't really do this. Since Continuous Monitoring is the only trigger, the only way to send a notification is for a new issue to be created that impacts one of their projects.

We could add a test notification endpoint, or automatically send out a test notification when their preferences are updated. But since it hasn't been done yet it shouldn't be in the docs.

Author: Gotcha, was wondering if we had a test that could be triggered. I'll update.

> ### Microsoft Teams Webhook Integration

Reviewer 3: These could be condensed into one section with links to setup for the different platforms being sub-items under `1.`

Reviewer 2: @mathew-horner looks updated, is this resolved?

1. Set up an incoming webhook in Microsoft Teams as described in [Microsoft Teams Webhook Setup](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook).
2. In your Phylum account, go to the 'Notifications' section.
3. Enter your Microsoft Teams webhook URL.
4. Perform a test to ensure the notifications are being sent correctly.

### Discord Webhook Integration

1. Create a Discord webhook following Discord's guidelines at [Discord Webhook Guide](https://discord.com/developers/docs/resources/webhook#execute-slackcompatible-webhook). Note that Discord support currently uses a Slack-compatible webhook format.
2. In Phylum, access the 'Notifications' section.
3. Paste your Discord webhook URL.
4. Perform a test to ensure the notifications are being sent correctly.

> ## Conclusion

Reviewer: this makes me think that adding a relevant icon next to the hook would not be so bad... unless using two of the three apps, or all of them, is highly unlikely

Reviewer 3: On the backend they have programmed Phulse to programatically figure out which type of URL it is. I think it makes more sense to configure it on the frontend, but each approach has its benefits. With this we can be agnostic about what type of webhook it is in the UI (the API probably needs to update a bit to reflect this, but that's not very high priority I don't think).

Reviewer 3: We could give them the option to add a label to each webhook. That would be quite easy to do.

> Setting up notifications through email or webhooks ensures you stay updated with your Phylum account activities. Follow these steps to integrate Slack, Microsoft Teams, or Discord with your Phylum account for real-time notifications.

Reviewer 3: I'd rewrite it like this, the second section leaves out email as a communication method but all the methods have been enumerated in the document anyways. No need to do it again, IMO.

```suggestion
Setting up notifications for emails or webhooks ensures you stay updated with your Phylum account activities. Follow these steps to receive notifications via your preferred communication method.
```
